File: Domain/Models/Enums/InputMethod.cs
using System.ComponentModel;

namespace Domain
{
    public enum InputMethod
    {
        [Description("Text Field")]
        TextField,

        [Description("Combo Box")]
        ComboBox,

        [Description("Editable Combo Box")]
        EditableComboBox,

        [Description("Date Picker")]
        DatePicker,

        [Description("Time Picker")]
        TimePicker,

        [Description("Radio Button")]
        RadioButton,

        [Description("Check Box")]
        CheckBox,
    }
}

File: Client/lib/models/Enum/InputMethod.dart
enum InputMethod {
  TextField,
  ComboBox,
  EditableComboBox,
  DatePicker,
  TimePicker,
  RadioButton,
  CheckBox,
}

extension InputMethodExtension on InputMethod {
  String get displayName {
    switch (this) {
      case InputMethod.TextField:
        return "Text Field";
      case InputMethod.ComboBox:
        return "Combo Box";
      case InputMethod.EditableComboBox:
        return "Editable Combo Box";
      case InputMethod.DatePicker:
        return "Date Picker";
      case InputMethod.TimePicker:
        return "Time Picker";
      case InputMethod.RadioButton:
        return "Radio Button";
      case InputMethod.CheckBox:
        return "Checkbox";
      default:
        return "Unknown";
    }
  }
}
